x-pack/elastic-agent/pkg/agent/application/filters/stream_checker.go

```diff
@@ -137,63 +133,88 @@ func StreamChecker(log *logger.Logger, ast *transpiler.AST) error {
-			return ErrInvalidIndex
+		if !matchesDatasetConstraints(datasetName) {
+			return ErrInvalidDataset
 		}
 	}
 
 	return nil
 }
 
-// The only two requirement are that it has only characters allowed in an Elasticsearch index name
-// and does NOT contain a `-`.
-func isValid(namespace string) bool {
-	return matchesIndexContraints(namespace) && !strings.Contains(namespace, "-")
-}
-
 // The only two requirement are that it has only characters allowed in an Elasticsearch index name
 // Index names must meet the following criteria:
+//	   Not longer than 100 bytes
 //     Lowercase only
 //     Cannot include \, /, *, ?, ", <, >, |, ` ` (space character), ,, #
+func matchesNamespaceContraints(namespace string) bool {
+	// length restriction is in bytes, not characters
+	if len(namespace) <= 0 || len(namespace) > 100 {
+		return false
+	}
+
+	// Lowercase only
+	if strings.ToLower(namespace) != namespace {
+		return false
+	}
+
+	// Cannot include \, /, *, ?, ", <, >, |, ` ` (space character), ,, #
+	if strings.ContainsAny(namespace, "\\/*?\"<>| ,#:") {
+		return false
+	}
+
+	return true
+}
+
+// matchesTypeConstraints fails for following rules. As type is first element of resulting index prefix restrictions need to be applied.
+//	   Not longer than 20 bytes
+//     Lowercase only
 //     Cannot start with -, _, +
-//     Cannot be . or ..
-func matchesIndexContraints(namespace string) bool {
-	// Cannot be . or ..
-	if namespace == "." || namespace == ".." {
+//     Cannot include \, /, *, ?, ", <, >, |, ` ` (space character), ,, #
+func matchesTypeConstraints(dsType string) bool {
+	// length restriction is in bytes, not characters
+	if len(dsType) <= 0 || len(dsType) > 20 {
 		return false
 	}
 
-	if len(namespace) <= 0 || len(namespace) > 255 {
+	if strings.ToLower(dsType) != dsType {
 		return false
 	}
 
-	// Lowercase only
-	if strings.ToLower(namespace) != namespace {
+	if strings.HasPrefix(dsType, "-") || strings.HasPrefix(dsType, "_") || strings.HasPrefix(dsType, "+") {
 		return false
 	}
 
-	// Cannot include \, /, *, ?, ", <, >, |, ` ` (space character), ,, #
-	if strings.ContainsAny(namespace, "\\/*?\"<>| ,#") {
+	if strings.ContainsAny(dsType, "\\/*?\"<>| ,#:") {
+		return false
+	}
+	return true
+}
+
+// matchesDatasetConstraints fails for following rules
+//	   Not longer than 100 bytes
+//     Lowercase only
+//     Cannot include \, /, *, ?, ", <, >, |, ` ` (space character), ,, #
+func matchesDatasetConstraints(dataset string) bool {
+	// length restriction is in bytes, not characters
+	if len(dataset) <= 0 || len(dataset) > 100 {
 		return false
 	}
 
-	// Cannot start with -, _, +
-	if strings.HasPrefix(namespace, "-") || strings.HasPrefix(namespace, "_") || strings.HasPrefix(namespace, "+") {
+	if strings.ToLower(dataset) != dataset {
 		return false
 	}
 
+	if strings.ContainsAny(dataset, "\\/*?\"<>| ,#:") {
```

Comment:
See above.

It looks like all 3 methods are identical besides the length limit?